src/internal/ErroneousOpenedFile.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of vfsStream.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package  bovigo\vfs
+ */
+
+namespace bovigo\vfs\internal;
+
+use bovigo\vfs\StreamWrapper;
+use const E_USER_WARNING;
+use function trigger_error;
+
+/**
+ * Decorator for vfsErronousFile to allow multiple instances of a file to be open.
+ *
+ * Decorates a regular opened file and triggers errors when configured to do so.
+ *
+ * @internal
+ */
+class ErroneousOpenedFile implements FileHandle
+{
+    /** @var  FileHandle */
+    private $openedFile;
+    /** @var string[] */
+    private $errorMessages;
+
+    /**
+     * @param string[] $errorMessages Formatted as [action => message], e.g. ['open' => 'error message']
+     */
+    public function __construct(FileHandle $openedFile, array $errorMessages)
+    {
+        $this->openedFile = $openedFile;
+        $this->errorMessages = $errorMessages;
+    }
+
+    /**
+     * locks file
+     *
+     * @see     https://github.com/mikey179/vfsStream/issues/6
+     * @see     https://github.com/mikey179/vfsStream/issues/40
+     *
+     * @param resource|StreamWrapper $resource
+     */
+    public function lock($resource, int $operation): bool
```

Comment:
Can anybody spot why phpcs complains about this with "Method \bovigo\vfs\vfsErroneousFile::lock() does not have parameter type hint nor @param annotation for its parameter $resource."? I even copied the whole block including function definition from `OpenedFile` where it doesn't complain, but that didn't shut up phpcs. Maybe I'm missing something obvious because I'm starting to get blind about the code?

Reply:
Because I was looking at the wrong file. 🤦🏻‍♂️